anomalib/deploy/inferencers/base.py

```diff
@@ -0,0 +1,172 @@
+"""Base Inferencer for Torch and OpenVINO."""
+
+# Copyright (C) 2020 Intel Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions
+# and limitations under the License.
+
+from abc import ABC, abstractmethod
+from pathlib import Path
+from typing import Dict, Optional, Tuple, Union, cast
+
+import cv2
+import numpy as np
+from omegaconf import DictConfig, OmegaConf
+from torch import Tensor
+
+from anomalib.data.utils import read_image
+from anomalib.utils.normalization.cdf import normalize as normalize_cdf
+from anomalib.utils.normalization.cdf import standardize
+from anomalib.utils.normalization.min_max import normalize as normalize_min_max
+from anomalib.utils.post_process import superimpose_anomaly_map
+
+
+class Inferencer(ABC):
+    """Abstract class for the inference.
+
+    This is used by both Torch and OpenVINO inference.
+    """
+
+    @abstractmethod
+    def load_model(self, path: Union[str, Path]):
+        """Load Model."""
+        raise NotImplementedError
+
+    @abstractmethod
+    def pre_process(self, image: np.ndarray) -> Union[np.ndarray, Tensor]:
+        """Pre-process."""
+        raise NotImplementedError
+
+    @abstractmethod
+    def forward(self, image: Union[np.ndarray, Tensor]) -> Union[np.ndarray, Tensor]:
+        """Forward-Pass input to model."""
+        raise NotImplementedError
+
+    @abstractmethod
+    def post_process(self, predictions: Union[np.ndarray, Tensor], meta_data: Optional[Dict]) -> np.ndarray:
```

Comment:
base inference task and torch inference tasks return only np.ndarray, while openvino inferencer returns Tuple[np.ndarray, float]. Design-wise, this is not correct, right? The signature should be the same. Is there a reason why only `np.ndarray` is returned here for anomaly_map, and not scores?